### PR TITLE
adds update test

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: update/provider/aws.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// TODO
+func (a *AWS) WaitForUpdate(nextVersion string) error {
+	return nil
+}
```
Comment (xh3b4sd) [inline update/provider/aws.go:123]: This has still to be implemented but we need to sort some prerequisites before. I would like to get this into master anyway so we have the general thing already. Later when we fixed this here we can start using it right away without keeping the PR stale for long.

File: update/spec.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	// act against. The test processes the following steps to ensure updates
+	// work.
+	//
+	//     - Lookup the current version bundle version.
+	//     - Lookup the next version bundle version.
+	//     - Stop the test if there is no current version bundle version.
+	//     - Stop the test if there is no next version bundle version.
+	//     - Update the guest cluster to next version.
+	//     - Wait for the guest cluster to be completely updated.
+	//
```
Comment (rossf7) [inline update/spec.go:19]: I really like that the interface describes the test. But it's maybe not obvious its here. Could the readme list the tests and link to the descriptions?

**Review by rossf7 — APPROVED** [review]
LGTM. Makes sense to get the structure into master.

**Review by MarcelMue — APPROVED** [review]
I like the direction taken here.
It would be nice to have an umbrella issue for this. The overall process is going to take a while, so an overview would be nice.

File: update/spec.go (hunk 2 above)
Comment (xh3b4sd) [reply update/spec.go:19]: Yeah lets polish the readme separately.

**Comment by xh3b4sd** [comment]
The umbrella for the update story is here: https://github.com/giantswarm/giantswarm/issues/1918.

**Pull request merged**